File: src/cdisc_library_api/specs/export_cdashig_scenario_variables_row.clj
(ns cdisc-library-api.specs.export-cdashig-scenario-variables-row
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def export-cdashig-scenario-variables-row-data
  {
   (ds/opt :Version) string?
   (ds/opt :Class) string?
   (ds/opt :Domain) string?
   (ds/opt :DataCollectionScenario) string?
   (ds/opt :VariableOrder) string?
   (ds/opt :CDASHIGVariable) string?
   (ds/opt :CDASHIGVariableLabel) string?
   (ds/opt :DRAFTCDASHIGDefinition) string?
   (ds/opt :QuestionText) string?
   (ds/opt :Prompt) string?
   (ds/opt :Type) string?
   (ds/opt :CDASHIGCore) string?
   (ds/opt :CaseReportFormCompletionInstructions) string?
   (ds/opt :SDTMIGTarget) (s/coll-of string?)
   (ds/opt :MappingInstructions) string?
   (ds/opt :ControlledTerminologyCodelistName) string?
   (ds/opt :ImplementationNotes) string?
   })

(def export-cdashig-scenario-variables-row-spec
  (ds/spec
    {:name ::export-cdashig-scenario-variables-row
     :spec export-cdashig-scenario-variables-row-data}))

File: src/main/scala/org/openapitools/client/api/ExportCdashigScenarioVariablesRow.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ExportCdashigScenarioVariablesRow._

case class ExportCdashigScenarioVariablesRow (
  version: Option[String],
`class`: Option[String],
domain: Option[String],
dataCollectionScenario: Option[String],
variableOrder: Option[String],
cDASHIGVariable: Option[String],
cDASHIGVariableLabel: Option[String],
dRAFTCDASHIGDefinition: Option[String],
questionText: Option[String],
prompt: Option[String],
`type`: Option[String],
cDASHIGCore: Option[String],
caseReportFormCompletionInstructions: Option[String],
sDTMIGTarget: Option[List[String]],
mappingInstructions: Option[String],
controlledTerminologyCodelistName: Option[String],
implementationNotes: Option[String])

object ExportCdashigScenarioVariablesRow {
  import DateTimeCodecs._

  implicit val ExportCdashigScenarioVariablesRowCodecJson: CodecJson[ExportCdashigScenarioVariablesRow] = CodecJson.derive[ExportCdashigScenarioVariablesRow]
  implicit val ExportCdashigScenarioVariablesRowDecoder: EntityDecoder[ExportCdashigScenarioVariablesRow] = jsonOf[ExportCdashigScenarioVariablesRow]
  implicit val ExportCdashigScenarioVariablesRowEncoder: EntityEncoder[ExportCdashigScenarioVariablesRow] = jsonEncoderOf[ExportCdashigScenarioVariablesRow]
}
